undertow/src/main/java/org/wildfly/extension/undertow/filters/ModClusterBalancerDefinition.java

```diff
@@ -43,147 +54,111 @@
  */
 class ModClusterBalancerDefinition extends SimpleResourceDefinition {
 
-    public static final ModClusterBalancerDefinition INSTANCE = new ModClusterBalancerDefinition();
-
-    public static final AttributeDefinition STICKY_SESSION_COOKIE = new SimpleAttributeDefinitionBuilder(Constants.STICKY_SESSION_COOKIE, ModelType.STRING)
-            .setRequired(false)
-            .setStorageRuntime()
-            .build();
-
-    public static final AttributeDefinition STICKY_SESSION_PATH = new SimpleAttributeDefinitionBuilder(Constants.STICKY_SESSION_PATH, ModelType.STRING)
-            .setRequired(false)
-            .setStorageRuntime()
-            .build();
-
-    public static final AttributeDefinition MAX_ATTEMPTS = new SimpleAttributeDefinitionBuilder(Constants.MAX_ATTEMPTS, ModelType.INT)
-            .setRequired(false)
-            .setStorageRuntime()
-            .build();
-
-    public static final AttributeDefinition WAIT_WORKER = new SimpleAttributeDefinitionBuilder(Constants.WAIT_WORKER, ModelType.INT)
-            .setRequired(false)
-            .setStorageRuntime()
-            .build();
-
-    public static final AttributeDefinition STICKY_SESSION = new SimpleAttributeDefinitionBuilder(Constants.STICKY_SESSION, ModelType.BOOLEAN)
-            .setRequired(false)
-            .setStorageRuntime()
-            .build();
-
-    public static final AttributeDefinition STICKY_SESSION_FORCE = new SimpleAttributeDefinitionBuilder(Constants.STICKY_SESSION_FORCE, ModelType.BOOLEAN)
-            .setRequired(false)
-            .setStorageRuntime()
-            .build();
-
-
-    public static final AttributeDefinition STICKY_SESSION_REMOVE = new SimpleAttributeDefinitionBuilder(Constants.STICKY_SESSION_REMOVE, ModelType.BOOLEAN)
-            .setRequired(false)
-            .setStorageRuntime()
-            .build();
-
-
-    private ModClusterBalancerDefinition() {
-        super(new Parameters(UndertowExtension.BALANCER, UndertowExtension.getResolver("handler", "mod-cluster", "balancer"))
-                .setRuntime());
-    }
-
-    @Override
-    public void registerChildren(ManagementResourceRegistration resourceRegistration) {
-        resourceRegistration.registerSubModel(ModClusterNodeDefinition.INSTANCE);
-        resourceRegistration.registerSubModel(ModClusterLoadBalancingGroupDefinition.INSTANCE);
-    }
-
-    @Override
-    public void registerAttributes(ManagementResourceRegistration resourceRegistration) {
-        resourceRegistration.registerReadOnlyAttribute(MAX_ATTEMPTS, new AbstractBalancerOperation() {
-
+    enum LoadBalancerMetric implements Metric<ModClusterStatus.LoadBalancer> {
+        STICKY_SESSION_COOKIE(Constants.STICKY_SESSION_COOKIE, ModelType.STRING) {
```

Comment:
So these were previously not annotated as access type `metrics`, but rather `read-only`. I think this is fine, but can we confirm this is the intention, since some of these are  not 'metrics' per se.

Reply:
Good point.  Let's leave that change for another time.  I'll revert this to normal attribute registration.

Reply:
Sounds good, thanks.

Reply:
Fixed.